extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ColumnPageReaderImpl.java

```diff
@@ -210,59 +222,100 @@ private IntBuffer readKeysFromDataPage(IntBuffer keyDest, int nullPlaceholder, S
                 }
             case DATA_PAGE_V2:
                 try (final InputStream in = channelsProvider.getInputStream(ch)) {
-                    readKeysFromPageV2(readV2Unsafe(in, channelContext), keyDest, nullPlaceholder, channelContext);
-                    return null;
+                    return readKeysFromPageV2(readV2Unsafe(in, channelContext), keyDest, nullPlaceholder,
+                            channelContext);
                 }
             default:
                 throw new IOException(String.format("Unexpected page of type %s of size %d", pageHeader.getType(),
                         pageHeader.getCompressed_page_size()));
         }
     }
 
-    private Object readDataPage(Object nullValue, SeekableByteChannel ch,
-            @NotNull SeekableChannelContext channelContext) throws IOException {
+    private Object readDataPage(
+            final Object nullValue,
+            final SeekableByteChannel ch,
+            @NotNull final SeekableChannelContext channelContext) throws IOException {
         switch (pageHeader.type) {
             case DATA_PAGE:
                 try (final InputStream in = channelsProvider.getInputStream(ch)) {
                     return readPageV1(readV1Unsafe(in, channelContext), nullValue, channelContext);
                 }
             case DATA_PAGE_V2:
                 try (final InputStream in = channelsProvider.getInputStream(ch)) {
-                    return readPageV2(readV2Unsafe(in, channelContext), nullValue);
+                    return readPageV2(readV2Unsafe(in, channelContext), nullValue, channelContext);
                 }
             default:
                 throw new IOException(String.format("Unexpected page of type %s of size %d", pageHeader.getType(),
                         pageHeader.getCompressed_page_size()));
         }
     }
 
-    private Encoding getEncoding(org.apache.parquet.format.Encoding encoding) {
+    private static Encoding getEncoding(final org.apache.parquet.format.Encoding encoding) {
         return org.apache.parquet.column.Encoding.valueOf(encoding.name());
     }
 
-    private int readRowCountFromPageV1(DataPageV1 page) {
+    private int readRowCountFromPageV1(final DataPageV1 page) {
         try {
             if (path.getMaxRepetitionLevel() != 0) {
                 // TODO - move away from page and use ByteBuffers directly
-                ByteBuffer bytes = page.getBytes().toByteBuffer();
+                final ByteBuffer bytes = page.getBytes().toByteBuffer();
                 bytes.order(ByteOrder.LITTLE_ENDIAN);
-                int length = bytes.getInt();
+                final int length = bytes.getInt();
                 return readRepetitionLevels(bytes.slice().limit(length));
             } else {
                 return page.getValueCount();
             }
-        } catch (IOException e) {
+        } catch (final IOException e) {
```

Comment:
It seems to me like:
1. The formatting might leave something to be desired.
2. We're inconsistent about catching this sort of thing. For instance, we don't do so on the key-reading path, at least not where I would expect.